plugins/ckan_operators/datastore_operator.py

```diff
@@ -249,3 +301,83 @@ def execute(self, context):
+    '''
+    Reads a JSON file and write the output into a CKAN datastore resource
+    
+    Expects as inputs:
+    - address - url of target ckan
+    - apikey - key needed to make authorized ckan calls
+    - resource_id - id of the resource that will receive this data
+    - data_path - location of the json data file
+    - fields_path - location of the data's fields, already in a CKAN-friendly format
+
+    All of the above, except the address and apikey, can be given with an actual value, or with a reference to a task_id and task_key that returns the value
+    '''
+    @apply_defaults
+    def __init__(
+        self,
+        address: str,
+        apikey: str,
+
+        resource_id: str = None,
+        resource_id_task_id: str = None,
+        resource_id_task_key: str = None,
+
+        data_path: str = None,
+        data_path_task_id: str = None,
+        data_path_task_key: str = None,
+
+        fields_path: str = None,
+        fields_path_task_id: str = None,
+        fields_path_task_key: str = None,
+
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.resource_id, self.resource_id_task_id, self.resource_id_task_key = resource_id, resource_id_task_id, resource_id_task_key
+        self.data_path, self.data_path_task_id, self.data_path_task_key = data_path, data_path_task_id, data_path_task_key
+        self.fields_path, self.fields_path_task_id, self.fields_path_task_key = fields_path, fields_path_task_id, fields_path_task_key
+        self.ckan = ckanapi.RemoteCKAN(apikey=apikey, address=address)
+
+
+    def execute(self, context):
+        # init task instance from context
+        ti = context['ti']
+
+        # assign important vars if provided from other tasks
+        if self.resource_id_task_id and self.resource_id_task_key:
+            self.resource_id = ti.xcom_pull(task_ids=self.resource_id_task_id)[self.resource_id_task_key]
+
+        if self.data_path_task_id and self.data_path_task_key:
+            self.data_path = ti.xcom_pull(task_ids=self.data_path_task_id)[self.data_path_task_key]
+
+        if self.fields_path_task_id and self.fields_path_task_key:
+            self.fields_path = ti.xcom_pull(task_ids=self.fields_path_task_id)[self.fields_path_task_key]
+
+
+        # get fields from file
+        with open(self.fields_path, "r") as f:
+            fields = json.load(f)
+            logging.info("Loaded the following fields from {}: {}".format( self.fields_path, fields ))
+
+        # populate that resource w data from the path provided
+        assert self.data_path, "Data path, or the filepath to the data to be inserted, must be provided!"
+        with open(self.data_path) as f:
+            data = json.load(f)
+        
+        logging.info("Data parsed from JSON file")
+        logging.info("Fields from fields file: " + str(fields))
+        logging.info("Fields from data file: " + str(data[0].keys()))
```

Comment:
I find multiple different logs easier to read than one with many lines (just because of how the log prefix with the timestamp lines up on the log page) but it's not a strong preference.

Im going to keep this as it is. That being said, I am noticing in these reviews that I put more lines of logs than might be needed, and I'll make a conscious effort to be more elegant / concise with my logging